packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js

```diff
@@ -120,19 +120,21 @@ const runQueriesForIds = ids => {
 
 const findDirtyIds = actions => {
   const state = store.getState()
-  return actions.reduce((dirtyIds, action) => {
-    const node = action.payload
+  return _.uniq(
```

Comment:
There were a ton of duplicates (though this wasn't too big of deal as our query queue removes duplicates).